### ISSUE TITLE
The keyword "type" is not treated as special when generating scala.js

### ISSUE THREAD

**Issue by espinhogr** [apollographql/apollo-tooling#1065]
Given a GraphQL type that looks like:
```
type Todo {
   id: String!
   type: String!
}
```

**Intended outcome:**
```
@scala.scalajs.js.native trait Todo extends scala.scalajs.js.Object {
      @scala.scalajs.js.annotation.JSName("id") val id: String
      @scala.scalajs.js.annotation.JSName("type") val `type`: String
}
```
Notice the declaration of the attribute "type" that is escaped with backquotes.

**Actual outcome:**
```
@scala.scalajs.js.native trait Todo extends scala.scalajs.js.Object {
      @scala.scalajs.js.annotation.JSName("id") val id: String
      @scala.scalajs.js.annotation.JSName("type") val type: String
}
```

**How to reproduce the issue:**
Create a GraphQL schema that has an object containing a field named "type" and try to generate the code for Scala.js .

**Versions**
apollo 2.5.3

**Comment by espinhogr** [comment]
I'm already working on it and I have a PR.

**Comment by trevor-scheer** [comment]
Fixed in #1066
